engines/illusions/duckman/duckman_videoplayer.cpp

```diff
@@ -69,7 +69,10 @@ void DuckmanVideoPlayer::update() {
 	} else if (_videoDecoder->needsUpdate()) {
 		const Graphics::Surface *frame = _videoDecoder->decodeNextFrame();
 		Graphics::Surface *backSurface = _vm->_screen->getBackSurface();
-		if (frame->format.bytesPerPixel == g_system->getScreenFormat().bytesPerPixel) {
+		if (!frame) {
```

Comment:
Should this condition be handled by `_videoDecoder->endOfVideo()` above?

Reply:
After taking a closer look, the issue appears to be that the last 14 out of 15 chunks in 00180003.AVI are blank, thus `_videoDecoder->decodeNextFrame()` returns a null pointer even when the end of the video has been reached, which results in a segmentation fault later.

Regardless, I modified the patch anyway, so that the timing should still be preserved instead of just skipping the video.